pytensor/tensor/rewriting/basic.py

```diff
@@ -941,6 +941,11 @@ def local_sum_make_vector(fgraph, node):
     elements = array.owner.inputs
     acc_dtype = node.op.acc_dtype
     out_dtype = node.op.dtype
+
+    # Skip rewrite when acc_dtype has higher precision than floatX
+    if is_an_upcast(config.floatX, acc_dtype):
```

Comment:
This is still too restrictive. floatx=32 does not mean float64 are forbidden, only that PyTensor won't introduce them automatically if they can be avoided. The user is still allowed to call `sum(dtype=float64)` or pass a float64 input in which case this check is too restrictive. There's already a float64 at one of the ends of this operation so it's fine for acc_dtype to be exposed. 

That's why my suggestion was quite more verbose.